scheduler/replica_scheduler.go

```diff
@@ -214,18 +225,18 @@ func (rcs *ReplicaScheduler) getDiskCandidates(nodeInfo map[string]*longhorn.Nod
 		}
 		if _, ok := usedNodes[nodeName]; !ok {
 			unusedNodes[nodeName] = node
-			if _, ok := usedZones[node.Status.Zone]; !ok {
-				unusedNodesInNewZones[nodeName] = node
-			}
 		}
 		if _, ok := usedZones[node.Status.Zone]; !ok {
-			nodesInUnusedZones[nodeName] = node
+			unusedNodesInUnusedZones[nodeName] = node
 		}
 	}
 
 	switch {
```

Comment:
NIT:  I still claim that since getDiskCandidatesFromNodes(unusedNodesInUnusedZones) is the first try in every case, it could be factored out and done before the switch, which might not even need to be executed.  Seems simpler to me, but I concede it's subjective.

Reply:
I forgot you said that, but I see what you mean. I will do as you suggest and see how it looks to other reviewers.